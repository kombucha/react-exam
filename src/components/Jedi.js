import React from "react";

const Jedi = ({ jedi }) => (
  <div>
    Jedi: id: {jedi.id}
    name: {jedi.name}
  </div>
);

Jedi.propTypes = {};

export default Jedi;
