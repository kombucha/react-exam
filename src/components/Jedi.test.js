import React from "react";
import Jedi from "./Jedi";
import { shallow } from "enzyme";

const testJedi = {
  id: 0,
  name: "Mace Windu",
};

it("should render properly", () => {
  const wrapper = shallow(<Jedi jedi={testJedi} />);
  expect(wrapper).toMatchSnapshot();
});
