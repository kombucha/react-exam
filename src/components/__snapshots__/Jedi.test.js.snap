// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jedi
    jedi={
        Object {
            "id": 0,
            "name": "Mace Windu",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Jedi: id: ",
        0,
        "name: ",
        "Mace Windu",
      ],
    },
    "ref": null,
    "rendered": Array [
      "Jedi: id: ",
      0,
      "name: ",
      "Mace Windu",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Jedi: id: ",
          0,
          "name: ",
          "Mace Windu",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Jedi: id: ",
        0,
        "name: ",
        "Mace Windu",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
